src/python/pants/rules/core/repl.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from dataclasses import dataclass
+from pathlib import PurePath
+
+from pants.base.build_root import BuildRoot
+from pants.engine.addressable import Addresses
+from pants.engine.console import Console
+from pants.engine.fs import Digest, DirectoryToMaterialize, Workspace
+from pants.engine.goal import Goal, GoalSubsystem
+from pants.engine.interactive_runner import InteractiveProcessRequest, InteractiveRunner
+from pants.engine.objects import union
+from pants.engine.rules import goal_rule
+from pants.engine.selectors import Get
+from pants.option.global_options import GlobalOptions
+from pants.util.contextutil import temporary_dir
+
+
+class ReplOptions(GoalSubsystem):
+  """Opens a REPL."""
+  name = 'repl2'
+
+
+class Repl(Goal):
+  subsystem_cls = ReplOptions
+
+
+@union
+class ReplDeterminer:
+  pass
```

Comment:
Maybe `ReplImplementation`?

Reply:
So, to be clear, possibly in a followup this will be an option like `./pants repl2 --implementation=ipython`? For now, we're punting on the specific option because there's only one possible implementation so it's unambigous? But this gives the infrastructure for more down the road?

Reply:
Yes that's what I had in mind with this type.